File: Love-Babbar-450-In-CSharp/03_string/02-is-palindrom.cs
using System;
using System.Collections.Generic;
using System.Text;
using Xunit;

namespace _03_string
{
    /*
        link: https://practice.geeksforgeeks.org/problems/palindrome-string0817/1
    */

    public class _02_is_palindrom
    {
        [Fact]
        public void palindrom_stringTest()
        {
            var ans = isPlaindrome1("racecar");
            ans = isPlaindrome1("racecar");
        }


        // ----------------------------------------------------------------------------------------------------------------------- //
        // method 1
        public int isPlaindrome1(string s)
        {
            // Your code goes here
            int len = s.Length;
            for (int i = 0; i < len / 2; i++)
            {
                if (s[i] != s[len - i - 1])
                {
                    return 0;
                }
            }
            return 1;
        }
    }
}

File: Love-Babbar-450-In-CPPTest/03_string/02-is-palindrom.cpp
/*
	link: https://practice.geeksforgeeks.org/problems/palindrome-string0817/1
*/

#include "CppUnitTest.h"
#include <iostream>
#include <algorithm>

using namespace std;
using namespace Microsoft::VisualStudio::CppUnitTestFramework;

namespace LoveBabbar450InCPPTest
{
	TEST_CLASS(StringClass)
	{
	public:

		TEST_METHOD(isPlaindromeTest)
		{
			std::string charM = "harhs";
			int age = 14;
			age = 55;
			std::string lastName = "<<charM <<singh";

		}
	};
}
// ----------------------------------------------------------------------------------------------------------------------- //
// method 1
class Solution {
public:


	int isPlaindrome(string s)
	{
		// Your code goes here
		int len = s.size();
		for (int i = 0;i < len / 2;i++) {
			if (s[i] != s[len - i - 1]) return 0;
		}
		return 1;
	}

};



// ----------------------------------------------------------------------------------------------------------------------- //
// method 2
class Solution2 {
public:


	int isPlaindrome(string s)
	{
		// Your code goes here
		string temp = s;
		reverse(temp.begin(), temp.end());
		return (temp == s);
	}

};